tools/cli/src/opt/kanidm.rs

```diff
@@ -1098,8 +1098,22 @@ pub enum Oauth2Opt {
     /// Disable legacy signing crypto on this oauth2 client. This is the default.
     #[clap(name = "disable-legacy-crypto")]
     DisableLegacyCrypto(Named),
-    #[clap(name = "prefer-short-username")]
-
+    /// Enable strict validation of redirect urls. Previously redirect urls only
+    /// validated the origin of the url matched. For public clients, if the client
+    /// site is vulnerable to an open redirection attack this can lead to token theft.
+    /// When enabled, redirect urls must match exactly.
+    #[clap(name = "enable-strict-redirect-url")]
```

Comment:
Were public clients affected *before* the change or after? It's not clear from the wording.

Reply:
It was a bit long winded so I cut it down, we explain the details elsewhere.